proposals/collection-literals.md

```diff
@@ -280,12 +303,27 @@ The existing rules for type inference (see [§11.6.3](https://github.com/dotnet/
 >   - **`V` is a single-dimensional array type `V₁[]` and `U` is a collection literal with element type `U₁`**
 >   - `V` is the type `V₁?` and `U` is the type `U₁`
 >   - `V` is a constructed type `C<V₁...Vₑ>` and `U` is a constructed type `C<U₁...Uₑ>`
->   - **`V` is a constructed type `C<V₁>` that is a constructible collection with element type `V₁`, and `U` is a collection literal with element type `U₁`**
+>   - **`V` is a constructible collection type `C<V₁>` with element type `V₁`, and `U` is a collection literal with element type `U₁`**
+>
 >   If any of these cases apply then an *exact inference* is made from each `Uᵢ` to the corresponding `Vᵢ`.
 
-_How do we recognize that `C<V₁>` is a constructible collection with element type `V₁`?_
+_How do we recognize that `C<V₁>` has element type `V₁`?_
 
 _Update to include inference from dictionary types._
+```c#
+var d = ["Alice": 42, "Bob": 43].AsDictionary(comparer);
```

Comment:
fwiw, I'm quite confused by this pattern as it has been proposed so far. We don't really do target typing of the receiver argument. The receiver in this example, I expect, has natural type `Dictionary<string, int>`. So I don't expect us to find the below `AsDictionary` method when we lookup members on it.

We could try to think of some way to *lookup members on types the CL could be target-typed to*, and finally choose a conversion and a member to access based on this. But it seems hard, and I also don't feel convinced it's worth it or that it will feel natural to users in the end.